urlparse4/cgurl.pyx

```diff
@@ -113,14 +119,36 @@ class SplitResultNamedTuple(tuple):
 
     __slots__ = ()  # prevent creation of instance dictionary
 
-    def __new__(cls, bytes url):
+    def __new__(cls, bytes url, decoded=False):
 
         cdef Parsed parsed
+        cdef Component scheme
+
+        if not ExtractScheme(url, len(url), &scheme):
+            """
+            TO DO:
+            What do we return here
+            """
+            return False
```

Comment:
will ``ParseStandardURL`` or ``ParsePathURL`` work here? If not, maybe raise an exception, and then in the python part we can either let it propogate, or catch it and use stdlib function instead?

Reply:
I will let it use stdlib function since we don't want any weird behavior from this library for now 😄